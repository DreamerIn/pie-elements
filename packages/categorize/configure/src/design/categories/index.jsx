import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import classNames from 'classnames';
import Category from './category';
import Header from '../header';

export class Categories extends React.Component {
  static propTypes = {
    imageSupport: PropTypes.shape({
      add: PropTypes.func.isRequired,
      delete: PropTypes.func.isRequired
    }),
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    columns: PropTypes.number.isRequired,
    onColumnsChange: PropTypes.func.isRequired,
    categories: PropTypes.array,
    onChange: PropTypes.func.isRequired,
    onDeleteChoice: PropTypes.func.isRequired,
    onAddChoice: PropTypes.func.isRequired,
    onAdd: PropTypes.func.isRequired,
    onDelete: PropTypes.func.isRequired
  };

  static defaultProps = {
    columns: 2
  };

  changeCategory = c => {
    const { onChange, categories } = this.props;

    const index = categories.findIndex(a => a.id === c.id);
    if (index !== -1) {
      categories.splice(index, 1, c);
      onChange(categories);
    }
  };

  render() {
    const {
      columns,
      classes,
      className,
      onColumnsChange,
      onDeleteChoice,
      categories,
      onAdd,
      onDelete,
      onAddChoice,
      imageSupport
    } = this.props;

    const holderStyle = {
      gridTemplateColumns: `repeat(${columns}, 1fr)`
    };

    return (
      <div className={classNames(classes.categories, className)}>
        <Header label="Categories" buttonLabel="ADD A CATEGORY" onAdd={onAdd} />
        <div className={classes.row}>
          <TextField
            label="Categories per row"
            type="number"
            inputProps={{
              min: 1,
              max: 4
            }}
            value={columns}
            onChange={onColumnsChange}
          />
        </div>
        <div className={classes.categoriesHolder} style={holderStyle}>
          {categories.map((category, index) => (
            <Category
              key={index}
              imageSupport={imageSupport}
              category={category}
              onChange={this.changeCategory}
              onDelete={() => onDelete(category)}
              onDeleteChoice={(choice, choiceIndex) =>
                onDeleteChoice(category, choice, choiceIndex)
              }
              onAddChoice={onAddChoice}
            />
          ))}
        </div>
      </div>
    );
  }
}

const styles = theme => ({
  categories: {
    marginBottom: theme.spacing.unit
  },
  categoriesHolder: {
    display: 'grid',
    gridRowGap: `${theme.spacing.unit}px`,
    gridColumnGap: `${theme.spacing.unit}px`
  },
  row: {
    display: 'grid',
    gridTemplateColumns: 'repeat(2, 1fr)',
    gridColumnGap: `${theme.spacing.unit}px`,
    alignItems: 'baseline',
    width: '100%',
    marginTop: theme.spacing.unit,
    marginBottom: 2 * theme.spacing.unit
  }
});

export default withStyles(styles)(Categories);
