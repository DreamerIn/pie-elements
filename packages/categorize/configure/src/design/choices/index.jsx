import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import Choice from './choice';
import Header from '../header';
import every from 'lodash/every';
import Config from './config';

export class Choices extends React.Component {
  static propTypes = {
    config: PropTypes.object.isRequired,
    onConfigChange: PropTypes.func.isRequired,
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    choices: PropTypes.array.isRequired,
    onChange: PropTypes.func.isRequired,
    onAdd: PropTypes.func.isRequired,
    onDelete: PropTypes.func.isRequired
  };

  static defaultProps = {};

  toggleRemoveAllTiles = () => {
    const { choices, onChange } = this.props;

    const allAtOne = this.allChoicesHaveCount(1);
    const update = choices.map(c => {
      c.categoryCount = allAtOne ? 0 : 1;
      return c;
    });
    onChange(update);
  };

  changeChoice = choice => {
    const { choices, onChange } = this.props;
    const index = choices.findIndex(h => h.id === choice.id);
    if (index !== -1) {
      choices.splice(index, 1, choice);
      onChange(choices);
    }
  };

  allChoicesHaveCount = count => {
    const { choices } = this.props;
    return every(choices, c => c.categoryCount === count);
  };

  render() {
    const {
      classes,
      className,
      choices,
      onAdd,
      onDelete,
      config,
      onConfigChange
    } = this.props;

    const categoryCountIsOne = this.allChoicesHaveCount(1);

    const choiceHolderStyle = {
      gridTemplateColumns: `repeat(${config.columns}, 1fr)`
    };
    return (
      <div className={classNames(classes.choices, className)}>
        <Header label="Choices" onAdd={onAdd} />

        <div className={classes.choiceHolder} style={choiceHolderStyle}>
          {choices.map((h, index) => (
            <Choice
              choice={h}
              correctResponseCount={h.correctResponseCount}
              key={index}
              onChange={this.changeChoice}
              onDelete={() => onDelete(h)}
            />
          ))}
        </div>
        <Config
          config={config}
          onChange={onConfigChange}
          categoryCountIsOne={categoryCountIsOne}
          onToggleCategoryCount={this.toggleRemoveAllTiles}
        />
      </div>
    );
  }
}
const styles = theme => ({
  row: {
    display: 'grid',
    gridColumnGap: `${theme.spacing.unit}px`,
    gridRowGap: `${theme.spacing.unit}px`,
    gridTemplateColumns: 'repeat(3, 1fr)',
    justifyContent: 'space-between',
    marginTop: theme.spacing.unit,
    marginBottom: theme.spacing.unit
  },
  choiceHolder: {
    paddingTop: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    display: 'grid',
    gridRowGap: `${theme.spacing.unit}px`,
    gridColumnGap: `${theme.spacing.unit}px`
  },
  choices: {
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2
  },
  label: {}
});
export default withStyles(styles)(Choices);
