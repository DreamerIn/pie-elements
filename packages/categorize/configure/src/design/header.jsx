import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import { AddButton } from './buttons';

export class Header extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    buttonLabel: PropTypes.string,
    onAdd: PropTypes.func.isRequired,
    label: PropTypes.string.isRequired
  };

  static defaultProps = {};
  render() {
    const { classes, className, onAdd, label, buttonLabel } = this.props;
    return (
      <div className={classNames(classes.header, className)}>
        <Typography variant="title" className={classes.title}>
          {label}
        </Typography>
        <AddButton onClick={onAdd} label={buttonLabel} />
      </div>
    );
  }
}
const styles = theme => ({
  header: {
    display: 'flex',
    justifyContent: 'space-between',
    marginBottom: 2 * theme.spacing.unit
  }
});
export default withStyles(styles)(Header);
