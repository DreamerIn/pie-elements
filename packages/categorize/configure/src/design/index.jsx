import React from 'react';
import PropTypes from 'prop-types';
import Typography from 'material-ui/Typography';
import { withStyles } from 'material-ui/styles';
import classNames from 'classnames';

export class Design extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string
  };
  static defaultProps = {};
  render() {
    const { classes, className } = this.props;
    return <div className={classNames(classes.design, className)} />;
  }
}
const styles = theme => ({
  design: {}
});
export default withStyles(styles)(Design);
