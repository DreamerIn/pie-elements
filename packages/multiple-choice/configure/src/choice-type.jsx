import PropTypes from 'prop-types';
import React from 'react';
import { TwoChoice } from '@pie-lib/config-ui';

export const ChoiceType = props => {
  let choiceProps = {
    header: props.header,
    defaultSelected: 'radio',
    value: props.value,
    onChange: props.onChange,
    one: {
      label: 'Radio',
      value: 'radio'
    },
    two: {
      label: 'Checkbox',
      value: 'checkbox'
    }
  };
  return <TwoChoice {...choiceProps} />;
};

export const KeyType = props => {
  let choiceProps = {
    header: props.header,
    defaultSelected: 'numbers',
    value: props.value,
    onChange: props.onChange,
    one: {
      label: 'Numbers',
      value: 'numbers'
    },
    two: {
      label: 'Letters',
      value: 'letters'
    }
  };
  return <TwoChoice {...choiceProps} />;
};

KeyType.propTypes = ChoiceType.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};
