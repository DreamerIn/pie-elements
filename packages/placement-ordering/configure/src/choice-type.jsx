import React from 'react';
import PropTypes from 'prop-types';
import { TwoChoice } from '@pie-lib/config-ui';

export const ChoiceType = props => {
  let choiceProps = {
    header: props.header,
    defaultSelected: props.value || 'vertical',
    value: props.value,
    onChange: props.onChange,
    one: props.one || {
      label: 'Vertical',
      value: 'vertical'
    },
    two: props.two || {
      label: 'Horizontal',
      value: 'horizontal'
    }
  };

  return <TwoChoice {...choiceProps} />;
};

ChoiceType.propTypes = {
  header: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  one: PropTypes.object,
  two: PropTypes.object,
};