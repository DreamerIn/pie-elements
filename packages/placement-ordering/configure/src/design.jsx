import {
  FeedbackConfig,
  FormSection,
  InputCheckbox,
  InputContainer,
  TwoChoice
} from '@pie-lib/config-ui';
import EditableHtml from '@pie-lib/editable-html';

import { get, set } from 'nested-property';

import ChoiceEditor from './choice-editor';
import PropTypes from 'prop-types';
import React from 'react';
import TextField from 'material-ui/TextField';
import cloneDeep from 'lodash/cloneDeep';
import debug from 'debug';
import { withStyles } from 'material-ui/styles';

const log = debug('@pie-element:placement-ordering:design');

class Design extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      allMoveOnDrag: false
    };

    this.applyUpdate = modelFn => {
      const { model, onModelChange } = this.props;
      const update = modelFn(cloneDeep(model));
      onModelChange(update);
    };

    this.changeHandler = (modelPath, valuePath) => {
      return value => {
        log('[changeHandler] value: ', value);
        const v = valuePath ? get(value, valuePath) : value;
        this.applyUpdate(model => {
          set(model, modelPath, v);
          return model;
        });
      };
    };

    this.onLayoutChange = layout => {
      this.applyUpdate(model => {
        model.choiceAreaLayout = layout;
        return model;
      });
    };

    this.onPlacementTypeChange = event => {
      const includePlacment = event.currentTarget.checked;
      this.applyUpdate(model => {
        model.placementType = includePlacment ? 'placement' : 'none';
        return model;
      });
    };

    this.onDefaultLangChange = defaultLang => {
      this.applyUpdate(model => {
        model.defaultLang = defaultLang;
        return model;
      });
    };

    this.onPromptChange = this.changeHandler('prompt');
    this.onChoiceAreaLabelChange = this.changeHandler(
      'choiceAreaLabel',
      'target.value'
    );
    this.onAnswerAreaLabelChange = this.changeHandler(
      'answerAreaLabel',
      'target.value'
    );
    this.onFeedbackChange = this.changeHandler('feedback');
    this.onShuffleChange = this.changeHandler('shuffle', 'target.checked');
    this.onShowOrderingChange = this.changeHandler(
      'showOrdering',
      'target.checked'
    );

    this.onChoiceEditorChange = (choices, correctResponse) => {
      const { model, onModelChange } = this.props;
      const update = cloneDeep(model);
      update.choices = choices;
      update.correctResponse = correctResponse;
      onModelChange(update);
    };
  }

  render() {
    const { model, classes, imageSupport } = this.props;
    const { allMoveOnDrag } = this.state;
    return (
      <div className={classes.design}>
        <div className={classes.row}>
          <TwoChoice
            className={classes.orientation}
            header={'Orientation'}
            value={model.choiceAreaLayout}
            onChange={this.onLayoutChange}
            one={{ label: 'vertical', value: 'vertical' }}
            two={{ label: 'horizontal', value: 'horizontal' }}
          />
        </div>
        <div className={classes.row}>
          <InputCheckbox
            label="Shuffle"
            checked={model.shuffle}
            onChange={this.onShuffleChange}
            aria-label="shuffle"
          />
          <InputCheckbox
            label="Include placement area"
            checked={model.placementType === 'placement'}
            onChange={this.onPlacementTypeChange}
            aria-label="include-placment"
          />
          <InputCheckbox
            disabled={model.placementType !== 'placement'}
            label="Numbered guides"
            checked={model.showOrdering}
            onChange={this.onShowOrderingChange}
            aria-label="shuffle"
          />
        </div>
        <InputContainer label="Prompt" className={classes.promptHolder}>
          <EditableHtml
            className={classes.prompt}
            markup={model.prompt}
            onChange={this.onPromptChange}
            imageSupport={imageSupport}
          />
        </InputContainer>

        <div className={classes.row}>
          <TextField
            className={classes.choiceLabel}
            label="Choice label"
            value={model.choiceAreaLabel}
            onChange={this.onChoiceAreaLabelChange}
            fullWidth
          />
          {model.placementType === 'placement' && (
            <TextField
              label="Answer label"
              value={model.answerAreaLabel}
              onChange={this.onAnswerAreaLabelChange}
              fullWidth
            />
          )}
        </div>
        <FormSection label="Choices">
          <ChoiceEditor
            correctResponse={model.correctResponse}
            choices={model.choices}
            onChange={this.onChoiceEditorChange}
            imageSupport={imageSupport}
          />
        </FormSection>
        <FeedbackConfig
          feedback={model.feedback}
          onChange={this.onFeedbackChange}
          imageSupport={imageSupport}
        />
      </div>
    );
  }
}

Design.propTypes = {
  model: PropTypes.object.isRequired,
  onModelChange: PropTypes.func.isRequired
};

export default withStyles(theme => ({
  promptHolder: {
    width: '100%'
  },
  prompt: {
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit
  },
  row: {
    display: 'grid',
    gridAutoFlow: 'column',
    gridAutoColumns: '1fr',
    gridGap: '8px'
  },
  design: {
    paddingTop: '10px'
  },
  langControls: {
    marginTop: '0px',
    marginBottom: '0px'
  },
  choices: {
    marginTop: '20px',
    marginBottom: '20px'
  },
  orientation: {
    marginTop: '0px',
    marginBottom: '0px'
  }
}))(Design);
