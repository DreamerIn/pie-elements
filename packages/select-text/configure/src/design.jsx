import React from 'react';
import PropTypes from 'prop-types';
import TextField from 'material-ui/TextField';
import { withStyles } from 'material-ui/styles';
import cloneDeep from 'lodash/cloneDeep';
import { Tokenizer } from '@pie-lib/text-select';
import {
  InputCheckbox,
  InputContainer,
  NumberTextField,
  FeedbackConfig
} from '@pie-lib/config-ui';
import Chip from 'material-ui/Chip';
import debug from 'debug';

const log = debug('@pie-element:select-text:configure');

export class Design extends React.Component {
  static propTypes = {
    model: PropTypes.object.isRequired,
    classes: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired,
    className: PropTypes.string
  };

  static defaultProps = {};

  changeText = event => {
    this.apply(u => {
      u.text = event.target.value;
      u.tokens = [];
    });
  };

  changeTokens = tokens => {
    this.apply(u => {
      u.tokens = tokens;
      const correctTokenCount = tokens.filter(t => t.correct).length;
      const max = isFinite(u.maxSelections) ? u.maxSelections : 0;
      u.maxSelections = Math.max(max, correctTokenCount);
    });
  };

  changeMaxSelections = (event, max) => {
    this.apply(u => (u.maxSelections = max));
  };

  changeHighlight = () => {
    this.apply(u => (u.highlightChoices = !u.highlightChoices));
  };

  apply = fn => {
    const { onChange, model } = this.props;
    const update = cloneDeep(model);
    fn(update);
    onChange(update);
  };

  changeFeedback = feedback => {
    this.apply(u => (u.feedback = feedback));
  };

  render() {
    const { model, classes, className } = this.props;

    log('[render] maxSelections:', model.maxSelections);

    return (
      <div className={className}>
        <TextField
          label="Content"
          className={classes.input}
          multiline
          value={model.text}
          onChange={this.changeText}
        />

        <InputCheckbox
          label={'Highlight choices'}
          checked={model.highlightChoices}
          onChange={this.changeHighlight}
        />

        <InputContainer label={'Tokens'} className={classes.tokenizerContainer}>
          <Tokenizer
            className={classes.tokenizer}
            text={model.text}
            tokens={model.tokens}
            onChange={this.changeTokens}
          />
        </InputContainer>
        <Chip
          label={`Selections Available: ${model.tokens.length}`}
          className={classes.chip}
        />
        <Chip
          label={`Correct Answers: ${
            model.tokens.filter(t => t.correct).length
          }`}
          className={classes.chip}
        />
        <NumberTextField
          min={0}
          label={'Selection count (0:any)'}
          max={model.tokens.length}
          value={model.maxSelections}
          onChange={this.changeMaxSelections}
        />

        <FeedbackConfig
          feedback={model.feedback}
          onChange={this.changeFeedback}
        />
      </div>
    );
  }
}
export default withStyles(theme => ({
  tokenizerContainer: {
    paddingRight: 0,
    marginRight: 0,
    '&:after': {
      position: 'absolute',
      bottom: 0,
      left: 0,
      right: 0,
      height: '1px',
      content: '""',
      backgroundColor: theme.palette.primary.main
    },
    marginBottom: theme.spacing.unit
  },
  chip: {
    marginRight: theme.spacing.unit * 2
  },
  input: {
    width: '100%',
    paddingBottom: theme.spacing.unit * 3
  },
  tokenizer: {
    marginTop: theme.spacing.unit * 2
  },
  mainOpts: {
    width: '100%',
    justifyContent: 'space-between',
    display: 'flex',
    alignItems: 'baseline'
  }
}))(Design);
