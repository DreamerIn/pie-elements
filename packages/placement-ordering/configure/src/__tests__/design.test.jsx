import { shallow } from 'enzyme';
import React from 'react';
import _ from 'lodash';

import {
  TwoChoice,
  FeedbackConfig,
  FormSection,
  InputCheckbox,
  InputContainer,
} from '@pie-lib/config-ui';
import { get, set } from 'nested-property';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import { Design } from '../design';
import defaultValues from '../defaults';
import { ChoiceType } from '../choice-type';

jest.mock('@pie-lib/config-ui', () => ({
  TwoChoice: props => (<div/>),
  FeedbackConfig: props => (<div/>),
  FormSection: props => (<div/>),
  InputCheckbox: props => (<div/>),
  InputContainer: props => (<div/>),
}));

jest.mock('nested-property', () => ({
  set: jest.fn((model, modelPath, v) => {
    model[modelPath] = v;

    return model;
  }),
  get: jest.fn((value, valuePath) => {
    switch (valuePath) {
      case 'value':
        return value[valuePath];
      case 'taget.value':
        return value.target.value;
      default:
        return value;
    }
  }),
}));

describe('Placement Ordering', () => {
  let updateModel;
  let model;

  beforeEach(() => {
    updateModel = jest.fn();
    model = _.cloneDeep(defaultValues);
  });

  describe('snapshot', () => {
    it ('renders all default items', () => {
      const wrapper = shallow(
        <Design
          model={model}
          classes={{}}
          className={'foo'}
          updateModel={updateModel}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });

    it ('renders custom items', () => {
      model.configure.editableItemStem = false;
      model.configure.settingsRemoveTileAfterPlacing = true;

      const wrapper = shallow(
        <Design
          model={model}
          classes={{}}
          className={'foo'}
          updateModel={updateModel}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    let w;
    let modelFn;

    beforeEach(() => {
      const wrapper = (extras) => {
        const defaults = {
          classes: {},
          className: 'className',
          updateModel,
          model
        };
        const props = { ...defaults, ...extras };

        return shallow(<Design {...props} />);
      };

      w = wrapper();
    });
    
    describe('applyUpdate', () => {
      beforeEach(() => {
        modelFn = jest.fn().mockReturnValue({
          ...model,
          itemStem: 'Updated Item Stem'
        });

        w.instance().applyUpdate(modelFn);
      });

      it('calls updateModel with updated item stem value', () => {
        expect(updateModel).toHaveBeenCalledWith({
          ...model,
          itemStem: 'Updated Item Stem'
        });
      });
    });

    describe('changeHandler', () => {
      let change;

      beforeEach(() => {
        change = (modelPath, valuePath, value) => {
          const onChoiceAreaLabelChange = w.instance().changeHandler(modelPath, valuePath);

          onChoiceAreaLabelChange(value);
        }
      });

      it('calls updateModel with updated item stem', () => {
        const modelPath = 'itemStem';
        const value = 'Updated Choice Area Label';

        change(modelPath, undefined, value);

        expect(updateModel).toHaveBeenCalledWith({
          ...model,
          [modelPath]: value
        });
      });

      it('calls updateModel with updated choice area label', () => {
        const modelPath = 'choiceLabel';
        const valuePath = 'value';
        const value = 'Updated Choice Area Label';

        change(modelPath, valuePath, { value });

        expect(updateModel).toHaveBeenCalledWith({
          ...model,
          [modelPath]: value
        });
      });

      it('calls updateModel with updated answer area label', () => {
        const modelPath = 'targetLabel';
        const valuePath = 'value';
        const value = 'Updated Answer Area Label';

        change(modelPath, valuePath, { value });

        expect(updateModel).toHaveBeenCalledWith({
          ...model,
          [modelPath]: value
        });
      });

      it('calls updateModel with updated feedback', () => {
        const modelPath = 'feedback';
        const value = {};

        change(modelPath, undefined, value);

        expect(updateModel).toHaveBeenCalledWith({
          ...model,
          [modelPath]: value
        });
      });
    });

    describe('onPromptChange', () => {
      it('calls update model when prompt changes', () => {
        let newItemStem = 'New item stem';

        w.instance().onPromptChange(newItemStem);

        expect(updateModel).toBeCalledWith({
          ...model,
          itemStem: newItemStem
        });
      });
    });

    describe('onChoiceAreaLabelChange', () => {
      it('calls update model when choice area label changes', () => {
        let newChoiceAreaLabel = 'New Choice Area Label';

        w.instance().onChoiceAreaLabelChange(newChoiceAreaLabel);

        expect(updateModel).toBeCalledWith({
          ...model,
          choiceLabel: newChoiceAreaLabel
        });
      });
    });

    describe('onAnswerAreaLabelChange', () => {
      it('calls update model when answer area label changes', () => {
        let newAnswerAreaLabel = 'New Answer Area Label';

        w.instance().onAnswerAreaLabelChange(newAnswerAreaLabel);

        expect(updateModel).toBeCalledWith({
          ...model,
          targetLabel: newAnswerAreaLabel
        });
      });
    });

    describe('onFeedbackChange', () => {
      it('calls update model when feedback changes', () => {
        let newFeedback = {
          correct: {
            type: 'custom',
            custom: 'CORRECT'
          },
          incorrect: {
            type: 'custom',
            custom: 'INCORRECT'
          },
          partial: {
            type: 'custom',
            custom: 'PARTIAL'
          }
        };

        w.instance().onFeedbackChange(newFeedback);

        expect(updateModel).toBeCalledWith({
          ...model,
          feedback: newFeedback
        });
      });
    });

    describe('onChoiceEditorChange', () => {
      it('calls update model when choices array changes', () => {
        let choices = [
          { id: 'c1', label: 'Choice 1' },
          { id: 'c2', label: 'Choice 2' }
        ];
        let correctResponse = [
          { id: 'c2' },
          { id: 'c1' }
        ];

        w.instance().onChoiceEditorChange(choices, correctResponse);

        expect(updateModel).toBeCalledWith({
          ...model,
          choices,
          correctResponse
        });
      });
    });

    describe('onSettingsChange', () => {
      it('calls update model when settings switches are toggled', () => {
        const formControls = w.find(FormControlLabel);

        formControls.forEach(fC => {
          fC.props().control.props.onChange(fC.props());

          expect(updateModel).toBeCalled();
        });
      });

      it('calls update model when orientation setting is changed', () => {
        const choiceType = w.find(ChoiceType);
        choiceType.props().onChange('horizontal');

        expect(updateModel).toBeCalledWith({
          ...model,
          choicesOrientation: 'horizontal'
        });
      });
    });
  })
});



