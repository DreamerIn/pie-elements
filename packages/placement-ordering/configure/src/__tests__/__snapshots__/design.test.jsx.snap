// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placement Ordering snapshot renders all default items 1`] = `
<div>
  <div>
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={
        <WithStyles(Switch)
          checked={false}
          onChange={[Function]}
          value="checkedA"
        />
      }
      key="0"
      label="Choice label"
      labelPlacement="start"
      style={
        Object {
          "justifyContent": "flex-end",
          "margin": 0,
          "width": "33%",
        }
      }
    />
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={
        <WithStyles(Switch)
          checked={false}
          onChange={[Function]}
          value="checkedA"
        />
      }
      key="6"
      label="Shuffle Choices"
      labelPlacement="start"
      style={
        Object {
          "justifyContent": "flex-end",
          "margin": 0,
          "width": "33%",
        }
      }
    />
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={
        <WithStyles(Switch)
          onChange={[Function]}
          value="checkedA"
        />
      }
      key="7"
      label="Partial Scoring"
      labelPlacement="start"
      style={
        Object {
          "justifyContent": "flex-end",
          "margin": 0,
          "width": "33%",
        }
      }
    />
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={
        <WithStyles(Switch)
          checked={true}
          onChange={[Function]}
          value="checkedA"
        />
      }
      key="1"
      label="Placement Area"
      labelPlacement="start"
      style={
        Object {
          "justifyContent": "flex-end",
          "margin": 0,
          "width": "33%",
        }
      }
    />
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={
        <WithStyles(Switch)
          onChange={[Function]}
          value="checkedA"
        />
      }
      key="2"
      label="Numbered Guides"
      labelPlacement="start"
      style={
        Object {
          "justifyContent": "flex-end",
          "margin": 0,
          "width": "33%",
        }
      }
    />
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={
        <WithStyles(Switch)
          checked={false}
          onChange={[Function]}
          value="checkedA"
        />
      }
      key="3"
      label="Enable Images"
      labelPlacement="start"
      style={
        Object {
          "justifyContent": "flex-end",
          "margin": 0,
          "width": "33%",
        }
      }
    />
    <ChoiceType
      header="Orientation"
      key="5"
      onChange={[Function]}
      style={
        Object {
          "width": "100%",
        }
      }
      value="vertical"
    />
  </div>
  <FormSection
    label="Ordering"
  >
    <InputContainer
      label="ITEM STEM"
    >
      <EditableHtml
        markup="Question Prompt goes here"
        onChange={[Function]}
      />
    </InputContainer>
  </FormSection>
  <FormSection
    label="Define Choices"
  >
    <div>
      <InputContainer
        label="ANSWER LABEL"
      >
        <EditableHtml
          markup="Answer Label goes here"
          onChange={[Function]}
        />
      </InputContainer>
    </div>
    <InputContainer
      label="CHOICES"
    >
      <WithStyles(ChoiceEditor)
        choices={
          Array [
            Object {
              "id": "c1",
              "label": "Choice 1",
            },
          ]
        }
        correctResponse={
          Array [
            Object {
              "id": "c1",
            },
          ]
        }
        disableImages={true}
        onChange={[Function]}
      />
    </InputContainer>
  </FormSection>
  <FeedbackConfig
    onChange={[Function]}
  />
</div>
`;

exports[`Placement Ordering snapshot renders custom items 1`] = `
<div>
  <div>
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={
        <WithStyles(Switch)
          checked={false}
          onChange={[Function]}
          value="checkedA"
        />
      }
      key="0"
      label="Choice label"
      labelPlacement="start"
      style={
        Object {
          "justifyContent": "flex-end",
          "margin": 0,
          "width": "33%",
        }
      }
    />
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={
        <WithStyles(Switch)
          checked={false}
          onChange={[Function]}
          value="checkedA"
        />
      }
      key="6"
      label="Shuffle Choices"
      labelPlacement="start"
      style={
        Object {
          "justifyContent": "flex-end",
          "margin": 0,
          "width": "33%",
        }
      }
    />
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={
        <WithStyles(Switch)
          onChange={[Function]}
          value="checkedA"
        />
      }
      key="7"
      label="Partial Scoring"
      labelPlacement="start"
      style={
        Object {
          "justifyContent": "flex-end",
          "margin": 0,
          "width": "33%",
        }
      }
    />
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={
        <WithStyles(Switch)
          checked={true}
          onChange={[Function]}
          value="checkedA"
        />
      }
      key="1"
      label="Placement Area"
      labelPlacement="start"
      style={
        Object {
          "justifyContent": "flex-end",
          "margin": 0,
          "width": "33%",
        }
      }
    />
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={
        <WithStyles(Switch)
          onChange={[Function]}
          value="checkedA"
        />
      }
      key="2"
      label="Numbered Guides"
      labelPlacement="start"
      style={
        Object {
          "justifyContent": "flex-end",
          "margin": 0,
          "width": "33%",
        }
      }
    />
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={
        <WithStyles(Switch)
          checked={false}
          onChange={[Function]}
          value="checkedA"
        />
      }
      key="3"
      label="Enable Images"
      labelPlacement="start"
      style={
        Object {
          "justifyContent": "flex-end",
          "margin": 0,
          "width": "33%",
        }
      }
    />
    <WithStyles(WithFormControlContext(FormControlLabel))
      control={
        <WithStyles(Switch)
          checked={true}
          onChange={[Function]}
          value="checkedA"
        />
      }
      key="4"
      label="Remove tiles after placing"
      labelPlacement="start"
      style={
        Object {
          "justifyContent": "flex-end",
          "margin": 0,
          "width": "33%",
        }
      }
    />
    <ChoiceType
      header="Orientation"
      key="5"
      onChange={[Function]}
      style={
        Object {
          "width": "100%",
        }
      }
      value="vertical"
    />
  </div>
  <FormSection
    label="Define Choices"
  >
    <div>
      <InputContainer
        label="ANSWER LABEL"
      >
        <EditableHtml
          markup="Answer Label goes here"
          onChange={[Function]}
        />
      </InputContainer>
    </div>
    <InputContainer
      label="CHOICES"
    >
      <WithStyles(ChoiceEditor)
        choices={
          Array [
            Object {
              "id": "c1",
              "label": "Choice 1",
            },
          ]
        }
        correctResponse={
          Array [
            Object {
              "id": "c1",
            },
          ]
        }
        disableImages={true}
        onChange={[Function]}
      />
    </InputContainer>
  </FormSection>
  <FeedbackConfig
    onChange={[Function]}
  />
</div>
`;
